packages/dds/sequence/src/test/overlappingIntervalsIndex.spec.ts

```diff
@@ -0,0 +1,460 @@
+				{ start: 1, end: 1 },
+				{ start: 1, end: 3 },
+				{ start: 1, end: 6 },
+				{ start: 2, end: 3 },
+				{ start: 2, end: 5 },
+			]);
+
+			results = queryIntervalsByPositions(5, 5);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 6 },
+				{ start: 2, end: 5 },
+				{ start: 5, end: 6 },
+			]);
+		});
+
+		it("when adding duplicate intervals to the index", () => {
+			const interval4 = collection.add(1, 1, IntervalType.SlideOnRemove);
+			const interval5 = collection.add(2, 3, IntervalType.SlideOnRemove);
+			const interval6 = collection.add(1, 1, IntervalType.SlideOnRemove);
+			overlappingIntervalsIndex.add(interval4);
+			overlappingIntervalsIndex.add(interval5);
+			overlappingIntervalsIndex.add(interval6);
+
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 1, end: 1 },
+				{ start: 1, end: 1 },
+				{ start: 2, end: 3 },
+				{ start: 2, end: 3 },
+				{ start: 5, end: 6 },
+			]);
+		});
+
+		it("when removing intervals from the index", () => {
+			overlappingIntervalsIndex.remove(interval2);
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 5, end: 6 },
+			]);
+
+			// Add and remove duplicate intervals
+			const interval4 = collection.add(1, 1, IntervalType.SlideOnRemove);
+			overlappingIntervalsIndex.add(interval4);
+			overlappingIntervalsIndex.remove(interval1);
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 1, end: 1 },
+				{ start: 5, end: 6 },
+			]);
+
+			overlappingIntervalsIndex.remove(interval4);
+			results = queryIntervalsByPositions(1, 6);
+			assertSequenceIntervalsEqual(testSharedString, results, [{ start: 5, end: 6 }]);
+		});
+
+		it("when inserting or appending additional segments to the string", () => {
+			// Append a segment to the end of the string
+			testSharedString.insertText(7, "hijk");
+			const interval4 = collection.add(7, 9, IntervalType.SlideOnRemove);
+			overlappingIntervalsIndex.add(interval4);
+
+			/**
+			 * Visualization of intervals within the string after the last insertion:
+			 *
+			 *                  0 1 2 3 4 5 6 7 8 9 10
+			 *                  a b c d e f g h i j k
+			 *  interval1         ^
+			 *  interval2           [-]
+			 *  interval3                 [-]
+			 *  interval4                     [---]
+			 */
+
+			results = queryIntervalsByPositions(5, 7);
+			assertSequenceIntervalsEqual(testSharedString, results, [
+				{ start: 5, end: 6 },
+				{ start: 7, end: 9 },
+			]);
+
+			// Insert a segment in the middle of the string
+			testSharedString.insertText(3, "xx");
+
+			/**
+			 * Visualization of intervals within the string after the last insertion:
+			 *
+			 * interval2's endpoint slides backwards
```

Comment:
merge-tree terminology is that "forwards" means "higher index" and "backwards" means "lower index," which looks reversed from this convention. I'd stay consistent with that (I think all comments in this test need an update)